lib/mongo-stream.js

```diff
@@ -25,8 +27,19 @@ MongoStream.prototype._write = function (chunk, encoding, done) {
 
     dao.writeLog(data, function (err) {
       if (err) {
-        console.error('fh-bunyan failed to write log to mongodb', err);
-        console.error((err && err.stack) ? err.stack : 'no stack available');
+        if (err.hasOwnProperty('jse_cause') &&
+          err.jse_cause.jse_shortmsg == 'failed to connect to mongodb' &&
+          !locked) {                    //if the error is caused by the unavailability of mongodb
+          console.error('MongoDB Connection error', err);
```

Comment:
You're still printing out the exception every time, except just with a different message when not locked. You want to print out the regular (`fh-bunyan failed to write log...`) message the first time, and swallow everything after that.

As for the locking, I think A) need to change the name, as we're not really locking anything, and B) we should just be swallowing everything after the first error. In the case of Node Gateway, we'd still see a nasty stacktrace every minute or so, which isn't good. Perhaps we could set the flag on an error, and reset the flag if we ever log without error? For Node Gateway, we'd never successfully log, so the flag would never reset, and thus we'd never see the error message again. In the other apps though, we'd see the error if the database was up, but later went down.